paddle/fluid/memory/detail/memory_block.h

```diff
@@ -109,7 +99,7 @@ class MetadataCache {
   // used to manage CPU memory, the MemoryBlock::Desc resides at the beginning
   // of the memory block; when used to manage GPU memory, the
   // Meatadata resides in CPU memory indexed by cache_.
-  MemoryBlock::Desc load(const MemoryBlock* memory_block) const;
+  MemoryBlock::Desc* load_desc(const MemoryBlock* memory_block) const;
```

Comment:
Do not return non-const pointer inside const member function. Add overloaded functions instead.

Reply:
For this case, I think there's no need to add a const version for loading logic, I just change the function to a non-const part